src/lib.rs

```diff
@@ -452,17 +452,41 @@ impl Maildir {
 
     /// Copies a message from the current maildir to the targetted maildir.
     pub fn copy_to(&self, id: &str, target: &Maildir) -> std::io::Result<()> {
-        let src = self.path.join("cur").join(id);
-        let dst = target.path.join("cur").join(id);
-        fs::copy(src, dst)?;
+        let entry = self.find(id).ok_or_else(|| {
+            std::io::Error::new(std::io::ErrorKind::NotFound, "Mail entry not found")
+        })?;
+        let filename = entry
+            .path()
+            .file_name()
+            .ok_or_else(|| {
+                std::io::Error::new(
+                    std::io::ErrorKind::InvalidData,
+                    "Invalid mail entry file name",
+                )
+            })?
+            .to_string_lossy()
+            .to_string();
+        fs::copy(entry.path(), target.path().join("cur").join(filename))?;
```

Comment:
There should probably be a guard here to prevent copying to the same file (which, according to docs, will truncate the file to zero bytes). So if the source and destination paths are the same, return an error.

Reply:
Not sure about the error kind and message I used.
:heavy_check_mark: